### PR TITLE
[FLINK-25580][State Backends]Improve the logic of over fraction calcu…

### PR DESCRIPTION
**Pull request by Aitozi**
…late

## What is the purpose of the change

When I check the logic of rescale part of the rocksDB. I found that we do not have to multiply `overlapFraction` twice to evaluate the best score handle. So this pr is to remove the duplicate `* overlapFraction` and fix one typo.

## Verifying this change

I add an extra test to verify that: If two handles have the same number of intersect group, The higher `overlapFraction` will win.

### CODE REVIEW

**Comment by Aitozi** [comment]
cc @Myasuka Please help take a look when you are free .

**Review by Myasuka — COMMENTED** [review]
Thanks for the contribution!
1. Please fill the content of original JIRA ticket.
2. I suggest to split this PR to two commits, one is for `overlapFraction`, another could be a `hotfix` for typo.

**Comment by Aitozi** [comment]
@Myasuka Thanks for your suggestion, I have updated the jira description and split the commit. Please take a look when you are free.

File: flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBIncrementalCheckpointUtils.java [hunk 1]
```diff
@@ -46,28 +47,58 @@
...
+                return new Score(intersectGroup.getNumberOfKeyGroups(), overlapFraction);
+            };
+
+    /**
+     * Score of the state handle, intersect group range is compared first, and then compare the
+     * overlap fraction.
+     */
+    private static class Score implements Comparable<Score> {
+
+        public static Score MIN = new Score(Integer.MIN_VALUE, -1.0);
```
Comment (Myasuka) [inline flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBIncrementalCheckpointUtils.java:76]: (naming) StaticVariableName: Name 'MIN' must match pattern '^[a-z][a-zA-Z0-9]*_?$'.

**Review by Myasuka — CHANGES_REQUESTED** [review]
I think this PR looks good to me in general. Please fix the checkstyle problem.

File: flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBIncrementalCheckpointUtils.java (hunk 1 above)
Comment (Aitozi) [reply flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBIncrementalCheckpointUtils.java]: Fixed by make it `final`

**Comment by Aitozi** [comment]
@flinkbot run azure

**Review by Myasuka — APPROVED** [review]
LGTM.

**Pull request closed without merging**